### ISSUE TITLE
avoid root permissions?

### ISSUE THREAD

**Issue by fquirin** [meg768/rpi-ws281x#22]
Using the library requires root permissions aka `sudo` or you will get:
```
...$ node led-control.js
Can't open /dev/mem: Permission denied
Segmentation fault
```

My hope was that `/dev/gpiomem` is enough ([see this issue](https://github.com/beyondscreen/node-rpi-ws281x-native/issues/101)) so I made sure my user has access: `sudo usermod -a -G gpio $USER` but it doesn't seem to work. Still the same error.

Is it because of PWM? I found [this](https://github.com/dotnet/iot/blob/main/Documentation/raspi-pwm.md) but I'm not sure if it plays any role here.

**Comment by krisdahl** [comment]
Yeah so I've had same issue.  If you want PWM you need root access.  But you can use SPI on like port 10 and you won't need root access, just to be part of the gpio group.

**Comment by fquirin** [comment]
I think that was what I did. It was a bit messy to get it right though if I remember correctly :-|. Since I had 'spi-device' already as dependency I sticked with that  in the end.
